yorm/test/test_utilities.py

```diff
@@ -114,6 +114,26 @@ def test_exception_when_file_not_expected_but_found(self):
         with pytest.raises(exceptions.FileAlreadyExistsError):
             utilities.sync(self.Sample(), "sample.yml", existing=False)
 
+    def test_file_exts(self):
+        # Standards (argument, expection)
+        for ext in [('yml', 'yml'), ('yaml', 'yml'), ('json', 'json')]:
+            sample = utilities.sync(self.Sample(), "sample.{}".format(ext[0]))
+            assert ext[1] == sample.__mapper__.ext
+            sample.foo = 'bar'
+
+        # Default to yml
+        sample = utilities.sync(self.Sample(), "sample.custom")
+        assert 'yml' == sample.__mapper__.ext
+
+        # None
+        sample = utilities.sync(self.Sample(), "/path/with/no/ext")
```

Comment:
You'll want to create files in a temporary directory: https://pytest.org/latest/tmpdir.html